index.html

```diff
@@ -102,29 +109,79 @@ <h2>Items in cart</h2>
 
+        const cartListElement = document.getElementById('cart-list')
+
+        function addToCart(array, item) {
+          array.push({
+          product: item.product,
+          price: item.price,
+          count: 1,
+          })
+        }
+
         Array.from(document.querySelectorAll('.add-to-cart')).forEach(
           (button) => {
             button.addEventListener('click', () => {
               console.log('clicked add to cart')
+              
+              const typesOfItemsInCart = itemsInCart.length
 
-              itemsInCart.push({
+              const latestCartItem = {
                 product: button.dataset.product,
                 price: Number(button.dataset.price),
-              })
-
-              const itemsInCartCount = itemsInCart.length
+              }
+
+              // This will keep track if the prodcut has been added to itemsInCart
+              let isInCart = false
+
+              // Check if the cart is empty
+              if (itemsInCart.length === 0) {
+                addToCart(itemsInCart, latestCartItem)
+                isInCart = true
+              } else {
+                // Cart isn't empty, so is the same type of product already exist?
+                for (let i=0; i < typesOfItemsInCart; i++) {
+                  if (itemsInCart[i].product === latestCartItem.product) {
+                    itemsInCart[i].count++
+                    isInCart = true
+                    break
+                  }
+                }
+              }
+ 
+              // Adding new type of product to cart
+              if (isInCart === false) {
+                addToCart(itemsInCart, latestCartItem)
+                isInCart = true
+              }
 
               let cartTotalOnClick = 0
 
               itemsInCart.forEach((item) => {
-                cartTotalOnClick = cartTotalOnClick += item.price
+                cartTotalOnClick = cartTotalOnClick + ( item.price * item.count )
+              })
+
+              var cartCount = 0
+              itemsInCart.forEach((item) =>{
+                cartCount += item.count
               })
 
-              document.getElementById('cart-count').innerText = itemsInCartCount
+              document.getElementById('cart-count').innerText = cartCount
 
               document.getElementById(
                 'cart-total'
               ).innerText = cartTotalOnClick.toFixed(2)
+
+              console.log(itemsInCart)
+
+              // Removes the old cart view in HTML
+              removeAllChildNodes(cartListElement)
+              // New cart view by displaying every single items inside itemsInCart
+              itemsInCart.forEach((item) => {
+                const cartListItemElement = document.createElement('li')
+                cartListItemElement.innerText = `${item.product} ($${item.price} / per) x ${item.count} = ${(item.price * item.count).toFixed(2)}`
+                cartListElement.appendChild(cartListItemElement)
+              })
```

Comment:
This definitely works since we have a small set of items, but in the long run it's not very performant--you want to reduce for loops and reduce DOM manipulations when possible (see [JavasScript performance](https://www.w3schools.com/js/js_performance.asp)).

Instead of removing all child nodes each time, I'd add and update only as necessary. See my comments prefixed with **Performance improvement suggestion**.

```suggestion
```